docs/developer/gui.ipynb

```diff
@@ -0,0 +1,213 @@
+   "metadata": {},
+   "source": [
+    "2. Register ``domain-type``-``Parameter`` instance mapping to ``parameter_registry``.\n",
+    "\n",
+    "``Parameter`` and its subclasses have a class method ``from_type`` that helps to create a new parameter instance for a specific domain type.\n",
+    "\n",
+    "There are various parameter types already exist in ``ess.reduce.parameter`` module, but here we will show you how to make a new one.\n",
+    "\n",
+    "Once you know which ``Parameter`` to use for the specific ``domain-type``, \n",
+    "\n",
+    "you can register the new **parameter instance** to the **domain type** in the ``parameter_registry``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ess.reduce.parameter import parameter_registry, Parameter\n",
+    "\n",
+    "\n",
+    "class BinNumberParameter(Parameter): ...\n",
+    "\n",
+    "\n",
+    "parameter_registry[NumBins] = BinNumberParameter(\n",
+    "    name=NumBins.__name__,\n",
+    "    description='Number of bins in the histogram',\n",
+    "    default=NumBins(10),\n",
+    ")\n",
+    "\n",
+    "# You can also use ``from_type``(class method) helper to instantiate the parameter.\n",
+    "# parameter_registry[NumBins] = BinNumberParameter.from_type(NumBins)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "3. Register ``type[Parameter]`` - ``type[Widget]`` mapping to ``create_parameter_widget`` distpatch.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ess.reduce.widgets import create_parameter_widget\n",
+    "import ipywidgets as widgets\n",
+    "\n",
+    "\n",
+    "@create_parameter_widget.register(BinNumberParameter)\n",
+    "def scalar_parameter_widget(param: BinNumberParameter):\n",
+    "    return widgets.IntText(\n",
+    "        value=param.default, description=param.name, tooltip=param.description\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example UI from the tutorial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "from ess.reduce.ui import layout\n",
+    "# Select Workflow\n",
+    "_workflow_name, _workflow_obj = layout.children[0].children[0].options[0]\n",
+    "layout.children[0].children[0].value = _workflow_obj\n",
+    "del _workflow_name, _workflow_obj\n",
+    "# Set Output Parameter\n",
+    "layout.children[0].children[1].options[0]\n",
+    "layout.children[0].children[1].value = (layout.children[0].children[1].options[0],)\n",
+    "# Click ``Generate Parameters`` button\n",
+    "layout.children[0].children[3].children[0].click()\n",
+    "# Click ``Run Workflow`` button\n",
+    "layout.children[1].children[0].click()"
```

Comment:
Should we write something about running/deploying this through voila?

Reply:
Sure.
That can be in the `User-Guide` I guess...? 
This one is under `Development` Section.

Reply:
Actually, we don't have to do it until we implement the widget-building helper.
I mentioned it in the #68 
I posted this one first so that we can use this as a guide for implementing other widgets.